### PR TITLE
Completely overhaul the deploy scripts.

### PR DESCRIPTION
**Pull request by nsthorat**
We now have 2 methods as part of public API in Python.
1. `deploy_project`
Deploys an entire project directory to a HuggingFace space by default. Users can decide to upload a subset of datasets, and parameterize it depending on their needs.

```python
def deploy_project(
  hf_space: str,
  project_dir: Optional[str] = None,
  datasets: Optional[list[str]] = None,
  make_datasets_public: Optional[bool] = False,
  concepts: Optional[list[str]] = None,
  skip_cache_upload: Optional[bool] = False,
  skip_data_upload: Optional[bool] = False,
  skip_concept_upload: Optional[bool] = False,
  create_space: Optional[bool] = False,
  load_on_space: Optional[bool] = False,
  hf_space_storage: Optional[Union[Literal['small'], Literal['medium'], Literal['large']]] = None
)
```
2. `deploy_config`
Deploys a lilac configuration object to a HuggingFace space, loading it on the HuggingFace server.
```python
def deploy_config(
  hf_space: str,
  config: Config,
  create_space: Optional[bool] = False,
  hf_space_storage: Optional[Union[Literal['small'], Literal['medium'], Literal['large']]] = None
)
```

Both of these will always use the public pip package.

Public API also has the public CLI methods:
- `lilac deploy-project` which takes the same arguments as `deploy_project`.

For development staging demos, use:
`poetry run python -m scripts.deploy_staging`
```
def deploy_staging(hf_space: Optional[str] = None,
                 dataset: Optional[list[str]] = None,
                 concept: Optional[list[str]] = None,
                 skip_cache_upload: Optional[bool] = False,
                 skip_ts_build: Optional[bool] = False,
                 skip_data_upload: Optional[bool] = False,
                 create_space: Optional[bool] = False) -> None:
```

For publishing the public demo, we still use:
```
poetry run python -m scripts.deploy_demo \
  --project_dir=./demo_data \
  --config=./lilac_hf_space.yml \
  --hf_space=lilacai/lilac
```

All 4 scripts are very opinionated.

I have also added a guide, and an ipynb for demonstration.

### CODE REVIEW

**Review by dsmilkov — APPROVED** [review]
sorry for to click submit, so all comments were pending yesterday

File: lilac/deploy_project.py
[the remark this answers is not in the document]
Comment (nsthorat) [reply lilac/deploy_project.py:1]: done

File: lilac.yml
[the remark this answers is not in the document]
Comment (nsthorat) [reply lilac.yml:1]: done

File: development.md [hunk 1]
```diff
@@ -61,7 +61,7 @@ Details can be found at
 1. Deploy to your HuggingFace Space:
 
    ```
-   poetry run python -m scripts.deploy_hf \
+   poetry run python -m scripts.deploy_local \
```
[the remark this answers is not in the document]
Comment (nsthorat) [reply development.md]: done

**Pull request merged**